undefined reference to `dlopen' on ubuntu 20.04 with default gcc 9.4.0

```
git clone --recursive https://github.com/zeek/zeek
```
configure and make and make install finished.

But  `zkg install  zeek/spicy-plugin` with such faield
```
Running unit tests for "zeek/zeek/spicy-plugin"

error: failed to run tests for zeek/zeek/spicy-plugin: package build_command failed, see log in /home/jiamo/opt/var/lib/zkg/logs/spicy-plugin-build.log
```

The error is :
```
/usr/bin/ld: /home/jiamo/opt/lib/libhilti.a(driver.cc.o): in function `hilti::Driver::_symbol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/toolchain/src/compiler/driver.cc:470: undefined reference to `dlerror'
/usr/bin/ld: /home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/toolchain/src/compiler/driver.cc:472: undefined reference to `dlsym'
/usr/bin/ld: /home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/toolchain/src/compiler/driver.cc:476: undefined reference to `dlerror'
/usr/bin/ld: /home/jiamo/opt/lib/libhilti.a(library.cc.o): in function `hilti::rt::Library::symbol(std::basic_string_view<char, std::char_traits<char> >) const':
/home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:98: undefined reference to `dlerror'
/usr/bin/ld: /home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:100: undefined reference to `dlsym'
/usr/bin/ld: /home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:102: undefined reference to `dlerror'
/usr/bin/ld: /home/jiamo/opt/lib/libhilti.a(library.cc.o): in function `hilti::rt::Library::~Library()':
/home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:57: undefined reference to `dlclose'
/usr/bin/ld: /home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:59: undefined reference to `dlerror'
/usr/bin/ld: /home/jiamo/opt/lib/libhilti.a(library.cc.o): in function `hilti::rt::Library::open() const':
/home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:75: undefined reference to `dlsym'
/usr/bin/ld: /home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:67: undefined reference to `dlopen'
/usr/bin/ld: /home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/runtime/src/library.cc:70: undefined reference to `dlerror'
collect2: error: ld returned 1 exit status
make[2]: *** [src/compiler/CMakeFiles/spicyz.dir/build.make:88: bin/spicyz] Error 1
make[1]: *** [CMakeFiles/Makefile2:362: src/compiler/CMakeFiles/spicyz.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make: *** [Makefile:141: all] Error 2
```

`std::__cxx11::basic_string`. But we build zeek with `c++14` ?

The plugin `make VERBOSE=1` got
```
/usr/bin/c++  -O3 -g -DNDEBUG  -Wl,--export-dynamic -rdynamic CMakeFiles/spicyz.dir/bin/spicyz.cc.o CMakeFiles/zeek-compiler.dir/driver.cc.o CMakeFiles/zeek-compiler.dir/glue-compiler.cc.o  -o ../../bin/spicyz   -L/home/jiamo/opt/lib  -Wl,-rpath,/home/jiamo/opt/lib: -lhilti -lspicy 
/usr/bin/ld: /home/jiamo/opt/lib/libhilti.a(driver.cc.o): in function `hilti::Driver::_symbol(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/jiamo/engine/zeek/auxil/spicy/spicy/hilti/toolchain/src/compiler/driver.cc:470: undefined reference to `dlerror'
```

In zeek source `rm ./auxil/spicy/spicy/lib/libhilti.a` and   `make VERBOSE=1` got

```
make -f auxil/spicy/spicy/hilti/toolchain/CMakeFiles/hilti.dir/build.make auxil/spicy/spicy/hilti/toolchain/CMakeFiles/hilti.dir/build
make[2]: Entering directory '/home/jiamo/engine/zeek/build'
[ 35%] Linking CXX static library ../../lib/libhilti.a
cd /home/jiamo/engine/zeek/build/auxil/spicy/spicy/hilti/toolchain && /usr/bin/cmake -P CMakeFiles/hilti.dir/cmake_clean_target.cmake
cd /home/jiamo/engine/zeek/build/auxil/spicy/spicy/hilti/toolchain && /usr/bin/cmake -E cmake_link_script CMakeFiles/hilti.dir/link.txt --verbose=1
/usr/bin/ar qc ../../lib/libhilti.a  ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/clock.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/drain.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/error.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/handle.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/init.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/options.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/pipe.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/process.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/redirect.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/redirect.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/reproc.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/run.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/strv.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/utf.posix.c.o CMakeFiles/hilti-objects.dir/src/ast/builder/builder.cc.o CMakeFiles/hilti-objects.dir/src/ast/builder/type.cc.o CMakeFiles/hilti-objects.dir/src/ast/declarations/imported-module.cc.o CMakeFiles/hilti-objects.dir/src/ast/expression.cc.o CMakeFiles/hilti-objects.dir/src/ast/expressions/id.cc.o CMakeFiles/hilti-objects.dir/src/ast/location.cc.o CMakeFiles/hilti-objects.dir/src/ast/meta.cc.o CMakeFiles/hilti-objects.dir/src/ast/module.cc.o CMakeFiles/hilti-objects.dir/src/ast/node.cc.o CMakeFiles/hilti-objects.dir/src/ast/node_ref.cc.o CMakeFiles/hilti-objects.dir/src/ast/scope.cc.o CMakeFiles/hilti-objects.dir/src/ast/scope-lookup.cc.o CMakeFiles/hilti-objects.dir/src/ast/type.cc.o CMakeFiles/hilti-objects.dir/src/ast/types/enum.cc.o CMakeFiles/hilti-objects.dir/src/ast/types/integer.cc.o CMakeFiles/hilti-objects.dir/src/ast/types/tuple.cc.o CMakeFiles/hilti-objects.dir/src/base/code-formatter.cc.o CMakeFiles/hilti-objects.dir/src/base/logger.cc.o CMakeFiles/hilti-objects.dir/src/base/preprocessor.cc.o CMakeFiles/hilti-objects.dir/src/base/timing.cc.o CMakeFiles/hilti-objects.dir/src/base/type_erase.cc.o CMakeFiles/hilti-objects.dir/src/base/util.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/codegen.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/coercions.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/ctors.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/expressions.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/operators.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/statements.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/types.cc.o CMakeFiles/hilti-objects.dir/src/compiler/codegen/unpack.cc.o CMakeFiles/hilti-objects.dir/src/compiler/coercion.cc.o CMakeFiles/hilti-objects.dir/src/compiler/context.cc.o CMakeFiles/hilti-objects.dir/src/compiler/cxx/elements.cc.o CMakeFiles/hilti-objects.dir/src/compiler/cxx/formatter.cc.o CMakeFiles/hilti-objects.dir/src/compiler/cxx/linker.cc.o CMakeFiles/hilti-objects.dir/src/compiler/cxx/unit.cc.o CMakeFiles/hilti-objects.dir/src/compiler/driver.cc.o CMakeFiles/hilti-objects.dir/src/compiler/init.cc.o CMakeFiles/hilti-objects.dir/src/compiler/jit.cc.o CMakeFiles/hilti-objects.dir/src/compiler/optimizer.cc.o CMakeFiles/hilti-objects.dir/src/compiler/parser/driver.cc.o CMakeFiles/hilti-objects.dir/src/compiler/plugin.cc.o CMakeFiles/hilti-objects.dir/src/compiler/unit.cc.o CMakeFiles/hilti-objects.dir/src/compiler/visitors/coercer.cc.o CMakeFiles/hilti-objects.dir/src/compiler/visitors/normalizer.cc.o CMakeFiles/hilti-objects.dir/src/compiler/visitors/printer.cc.o CMakeFiles/hilti-objects.dir/src/compiler/visitors/renderer.cc.o CMakeFiles/hilti-objects.dir/src/compiler/visitors/resolver.cc.o CMakeFiles/hilti-objects.dir/src/compiler/visitors/scope-builder.cc.o CMakeFiles/hilti-objects.dir/src/compiler/visitors/validator.cc.o CMakeFiles/hilti-objects.dir/src/global.cc.o CMakeFiles/hilti-objects.dir/__/src/autogen/operators-implementations.cc.o CMakeFiles/hilti-objects.dir/__/src/autogen/config.cc.o CMakeFiles/hilti-objects.dir/__/src/autogen/__parser.cc.o CMakeFiles/hilti-objects.dir/__/src/autogen/__scanner.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/backtrace.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/configuration.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/context.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/debug-logger.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/exception.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/fiber.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/global-state.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/init.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/library.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/logging.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/main.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/type-info.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/address.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/bytes.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/integer.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/port.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/real.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/regexp.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/stream.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/string.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/types/time.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/src/util.cc.o ../runtime/CMakeFiles/hilti-rt-objects.dir/__/__/3rdparty/utf8proc/utf8proc.c.o ../../3rdparty/reproc/reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/ccl.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/dfa.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/dfa-interpreter-std.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/dfa-interpreter-min.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/jlocale.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/jrx.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/nfa.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/util.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/justrx/autogen/re-parse.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/justrx/autogen/re-scan.c.o ../../3rdparty/fiber/CMakeFiles/fiber.dir/src/fiber_asm_amd64_sysv.S.o ../../3rdparty/fiber/CMakeFiles/fiber.dir/src/fiber.c.o
/usr/bin/ranlib ../../lib/libhilti.a
```
I also see build the spicy-config with `-ldl`. So the `libhilti` is somthing internal lib don't need `-ldl` but `plugin` build need it ?
 
```
[ 40%] Linking CXX executable ../../bin/spicy-config
cd /home/jiamo/engine/zeek/build/auxil/spicy/spicy/spicy/toolchain && /usr/bin/cmake -E cmake_link_script CMakeFiles/spicy-config.dir/link.txt --verbose=1
/usr/bin/c++   -Wall -Wno-unused -Wno-register -Werror=vla -Wno-vla -O3 -g -DNDEBUG   -fuse-ld=gold -Wl,--export-dynamic -rdynamic CMakeFiles/spicy-config.dir/bin/spicy-config.cc.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/clock.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/drain.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/error.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/handle.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/init.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/options.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/pipe.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/process.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/redirect.posix.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/redirect.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/reproc.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/run.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/strv.c.o ../../3rdparty/reproc/reproc/CMakeFiles/reproc.dir/src/utf.posix.c.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/builder/builder.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/builder/type.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/declarations/imported-module.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/expression.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/expressions/id.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/location.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/meta.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/module.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/node.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/node_ref.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/scope.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/scope-lookup.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/type.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/types/enum.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/types/integer.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/ast/types/tuple.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/base/code-formatter.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/base/logger.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/base/preprocessor.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/base/timing.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/base/type_erase.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/base/util.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/codegen.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/coercions.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/ctors.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/expressions.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/operators.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/statements.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/types.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/codegen/unpack.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/coercion.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/context.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/cxx/elements.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/cxx/formatter.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/cxx/linker.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/cxx/unit.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/driver.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/init.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/jit.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/optimizer.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/parser/driver.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/plugin.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/unit.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/visitors/coercer.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/visitors/normalizer.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/visitors/printer.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/visitors/renderer.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/visitors/resolver.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/visitors/scope-builder.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/compiler/visitors/validator.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/src/global.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/__/src/autogen/operators-implementations.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/__/src/autogen/config.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/__/src/autogen/__parser.cc.o ../../hilti/toolchain/CMakeFiles/hilti-objects.dir/__/src/autogen/__scanner.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/backtrace.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/configuration.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/context.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/debug-logger.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/exception.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/fiber.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/global-state.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/init.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/library.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/logging.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/main.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/type-info.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/address.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/bytes.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/integer.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/port.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/real.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/regexp.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/stream.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/string.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/types/time.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/src/util.cc.o ../../hilti/runtime/CMakeFiles/hilti-rt-objects.dir/__/__/3rdparty/utf8proc/utf8proc.c.o ../../3rdparty/reproc/reproc++/CMakeFiles/reproc++.dir/src/reproc.cpp.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/ccl.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/dfa.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/dfa-interpreter-std.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/dfa-interpreter-min.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/jlocale.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/jrx.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/nfa.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/src/util.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/justrx/autogen/re-parse.c.o ../../3rdparty/justrx/CMakeFiles/jrx-objects.dir/justrx/autogen/re-scan.c.o ../../3rdparty/fiber/CMakeFiles/fiber.dir/src/fiber_asm_amd64_sysv.S.o ../../3rdparty/fiber/CMakeFiles/fiber.dir/src/fiber.c.o CMakeFiles/spicy-objects.dir/src/ast/hook.cc.o CMakeFiles/spicy-objects.dir/src/ast/types.cc.o CMakeFiles/spicy-objects.dir/src/ast/types/bitfield.cc.o CMakeFiles/spicy-objects.dir/src/ast/types/unit.cc.o CMakeFiles/spicy-objects.dir/src/ast/types/unit-items/field.cc.o CMakeFiles/spicy-objects.dir/src/ast/types/unit-items/switch.cc.o CMakeFiles/spicy-objects.dir/src/compiler/coercion.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/codegen.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/grammar.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/grammar-builder.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/parser-builder.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/parsers/literals.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/parsers/types.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/production.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/productions/look-ahead.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/productions/switch.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/productions/while.cc.o CMakeFiles/spicy-objects.dir/src/compiler/codegen/unit-builder.cc.o CMakeFiles/spicy-objects.dir/src/compiler/driver.cc.o CMakeFiles/spicy-objects.dir/src/compiler/init.cc.o CMakeFiles/spicy-objects.dir/src/compiler/parser/driver.cc.o CMakeFiles/spicy-objects.dir/src/compiler/plugin.cc.o CMakeFiles/spicy-objects.dir/src/compiler/visitors/normalizer.cc.o CMakeFiles/spicy-objects.dir/src/compiler/visitors/resolver.cc.o CMakeFiles/spicy-objects.dir/src/compiler/visitors/printer.cc.o CMakeFiles/spicy-objects.dir/src/compiler/visitors/scope-builder.cc.o CMakeFiles/spicy-objects.dir/src/compiler/visitors/validator.cc.o CMakeFiles/spicy-objects.dir/__/src/autogen/operators-implementations.cc.o CMakeFiles/spicy-objects.dir/__/src/autogen/config.cc.o CMakeFiles/spicy-objects.dir/__/src/autogen/__parser.cc.o CMakeFiles/spicy-objects.dir/__/src/autogen/__scanner.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/base64.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/driver.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/global-state.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/init.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/mime.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/parser.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/sink.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/unit-context.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/util.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/src/zlib.cc.o ../runtime/CMakeFiles/spicy-rt-objects.dir/__/__/3rdparty/libb64/src/cdecode.c.o ../runtime/CMakeFiles/spicy-rt-objects.dir/__/__/3rdparty/libb64/src/cencode.c.o  -o ../../bin/spicy-config  -Wl,-rpath,:::::::::::::::::::::: /usr/lib/x86_64-linux-gnu/libz.so ../../lib/libhilti.a -pthread -ldl -lrt 
```

And I also find  `./../bin/hilti-config` was linked by '-ldl' too.

--- a/cmake/FindSpicy.cmake
+++ b/cmake/FindSpicy.cmake
@@ -117,7 +117,9 @@ function (spicy_link_libraries lib)
                             ${SPICY_LIBRARY_DIRS_RUNTIME})
 
     if (SPICY_HAVE_TOOLCHAIN)
-        target_link_libraries(${lib} "${ARGN}" hilti spicy)
+        # In addition to libhilti and libspicy we also add their dependencies here
+        # so we can link against both shared libraries as well as static archives.
+        target_link_libraries(${lib} "${ARGN}" hilti spicy dl)
     endif ()
 endfunction ()
 

